pyaedt/edb_core/edb_data/siwave_simulation_setup_data.py

```diff
@@ -511,336 +779,289 @@ def refine_vias(self):
 
-    @property
-    def edb_sim_setup_info(self):
-        """EDB internal simulation setup object."""
-        return self._edb_sim_setup_info
+    Parameters
+    ----------
+    pedb : :class:`pyaedt.edb.Edb`
+        Inherited AEDT object.
+    edb_setup : :class:`Ansys.Ansoft.Edb.Utility.SIWaveSimulationSetup`
+        Edb object.
+    """
+
+    def __init__(self, pedb, edb_setup=None):
+        super().__init__(pedb, edb_setup)
+        self._edb = self._pedb
+        self._setup_type = "kSIwave"
+        self._sim_setup_info = None
 
     @pyaedt_function_handler()
-    def _update_setup(self):
-        self._edb_sim_setup = self._edb.edb_api.utility.utility.SIWaveSimulationSetup(self._edb_sim_setup_info)
-        if self.name in self._edb.setups:
-            self._edb.layout.cell.DeleteSimulationSetup(self.name)
-        self._edb.layout.cell.AddSimulationSetup(self._edb_sim_setup)
-        return True
+    def create(self, name=None):
+        """Create a SIwave SYZ setup.
 
-    @property
-    def dc_settings(self):
-        """Siwave DC settings.
+        Returns
+        -------
+        :class:`SiwaveDCSimulationSetup`
+        """
+        self._name = name
+        self._create(name)
+        self.set_si_slider(1)
+
+        return self
+
+    @pyaedt_function_handler
+    def get_configurations(self):
+        """Get SIwave SYZ simulation settings.
 
         Returns
         -------
-        :class:`pyaedt.edb_core.edb_data.siwave_simulation_setup_data.SiwaveDCAdvancedSettings`
+        dict
         """
-        return SiwaveDCAdvancedSettings(self)
+        return {
+            "pi_slider_postion": self.pi_slider_position,
+            "si_slider_postion": self.si_slider_position,
+            "use_custom_settings": self.use_si_settings,
+            "use_si_settings": self.use_si_settings,
+            "advanced_settings": self.advanced_settings.get_configurations(),
+        }
 
     @property
-    def frequency_sweeps(self):
-        """Get frequency sweep list."""
-        if self._sweep_data_list:
-            return self._sweep_data_list
-        self._sweep_data_list = {}
-        for i in list(self._edb_sim_setup_info.SweepDataList):
-            self._sweep_data_list[i.Name] = EdbFrequencySweep(self, None, i.Name, i)
-        return self._sweep_data_list
+    def advanced_settings(self):
+        """SIwave advanced settings class."""
+        return AdvancedSettings(self)
 
     @property
-    def name(self):
-        """Setup name."""
-        return self._edb_sim_setup_info.Name
-
-    @name.setter
-    def name(self, value):
-        """Set name of the setup."""
-        legacy_name = self._edb_sim_setup_info.Name
-        self._edb_sim_setup_info.Name = value
-        self._update_setup()
-        if legacy_name in self._edb.setups:
-            del self._edb._setups[legacy_name]
+    def get_sim_setup_info(self):
+        """Read simulation information from setup."""
```

Comment:
```suggestion
        """Get simulation information from the setup."""
```
Is there a strong reason to use "read" rather than "get"? Using the same verb as in the method is reinforcing--so if there is no need to use another verb, then don't.